### DISCUSSION TITLE
Feature ask: How to complete on host based on env in a docker container?

### DISCUSSION THREAD

**Discussion by QiangZiBro** [neoclide/coc.nvim#4246] [Q&A]
Hello, I love using coc.nvim. I use docker containers as my daily running environment, and I need Python/C++/CUDA env in the docker containers.
I would like to ask if it's possible to let coc.nvim complete from the env in a docker container? I thought about the potential solutions:
1. Install neovim+coc.nvim inside of docker container. This is too complicated and have to install every time when I use a container.
2. Copy the libraries (Python/C++/CUDA) from container to a directory in host, then let coc.nvim find from them. But I don't know how to do this yet.

Anybody has ideas?  Thank you!

**Comment by chemzqm** [comment]
It should be possible to make vim and coc.nvim work in your container, but I think you'd better code on your own machine and deploy the code to container, since the resources accessible by container is very limited, and coc.nvim, LS requires lots of resources.

**Reply by QiangZiBro** [reply]
Yes, I code on host machine, but the Python or C++ libraries are **only** inside of docker container, so I can't have any completion when coding. I'm thinking:
1. Is it possible to let coc.nvim complete the code based on the libraries inside of the container?
2. Or, I can just copy these libraries to host, is there a way to let coc.nvim find them?

**Reply by chemzqm** [reply]
You can copy the header files and configure clangd to recognize them.

**Reply by lithammer** [reply]
I think @QiangZiBro is requesting something similar to what you can do in VS Code.

https://code.visualstudio.com/docs/remote/containers-tutorial
https://code.visualstudio.com/docs/remote/attach-container

**Reply by QiangZiBro** [reply]
> I think @QiangZiBro is requesting something similar to what you can do in VS Code.
> 
> https://code.visualstudio.com/docs/remote/containers-tutorial https://code.visualstudio.com/docs/remote/attach-container

Yeah, it can be done with some mordern IDEs, curious about how can it be done in nvim elegantly

**Reply by QiangZiBro** [reply]
> You can copy the header files and configure clangd to recognize them.

Great, one more question, how to let coc.nvim find python packages in a custom directory?

**Reply by chemzqm** [reply]
Check the document of language server

**Comment by dty2** [comment]
@QiangZiBro  
Hey, man, did you solve the problem?
How do you do it? ：)
